### PR TITLE
Make popups and text fields nicer to interact with on Android

### PR DESCRIPTION
**Pull request by Azzurite**
I'm really proud of this one :)

Before:

https://user-images.githubusercontent.com/2777158/174455985-c4175460-e96a-4da3-a5c2-64c659c12dec.mp4

After:

https://user-images.githubusercontent.com/2777158/174455991-b0dc191a-6687-42db-a7ef-6c8a267aa257.mp4

I'm going to add a few comments as a review beside the code in a sec.

### CODE REVIEW

File: core/src/com/unciv/ui/utils/GeneralPlatformSpecificHelpers.kt [hunk 1]
```diff
@@ -35,4 +33,10 @@ interface GeneralPlatformSpecificHelpers {
      * @return true if the throwable was handled.
      */
     fun handleUncaughtThrowable(ex: Throwable): Boolean = false
+
+    /**
+     * Adds platform-specific improvements to the given text field, making it nicer to interact with on this platform.
+     */
+    fun addImprovements(textField: TextField): TextField = textField
```
[the remark this answers is not in the document]
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/utils/GeneralPlatformSpecificHelpers.kt:40]: What, Farms or Lumbermills on a TextField?

File: .idea/inspectionProfiles/Project_Default.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="000ce796-10c4-3e92-bd42-fa6f0bd6ec8b" enabled="true" level="ERROR" enabled_by_default="true" />
+    <inspection_tool class="SSBasedInspection" enabled="true" level="WARNING" enabled_by_default="true">
+      <searchConfiguration name="Raw TextField usage" description="We want to add some extra behavior to all text fields. This behavior will not be added by instantiating a TextField directly, so use our general factory function UncivTextField.createTextField instead." suppressId="UNCIV_RAW_TEXTFIELD" problemDescriptor="Use UncivTextField.createTextField instead!" text="TextField($Params$)" recursive="true" caseInsensitive="true" type="Kotlin" pattern_context="default">
```
Comment (SomeTroglodyte) [inline .idea/inspectionProfiles/Project_Default.xml:6]: 1 attribute per line? "caseInsensitive=true"?

**Review by SomeTroglodyte — APPROVED** [review]
Lots of work for something Gdx should be managing. Wow. I'll approve based on trust, not testing. I do Dislike top-level fields and methods very much, and UncivTextFilletMignon would look perfectly clean as subclass... But I know you're a pre-.net Visual Basic fan... so :shrug:

File: .idea/inspectionProfiles/Project_Default.xml (hunk 2 above)
Comment (SomeTroglodyte) [reply .idea/inspectionProfiles/Project_Default.xml:6]: OK, I thought you had to code the xml y'self...

**Comment by SomeTroglodyte** [comment]
> why don't you dislike top-level classes

I **do** - but [kotlin](https://arturdryomov.dev/posts/namespacing-in-kotlin/#the-decision) is [namespace](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/namespace)-crippled.

**Comment by MindaugasRumsa51** [comment]
i recommend change string:
"Are you sure you want to delete this save?"
to this string:
"Are you sure you want to delete this save?".tr()+"\n"+selectedSave

**Comment by touhidurrr** [comment]
Wow. The problem of typing in-game in most games is so common that we even most of the time just take it granted.

And yet someone found a good solution for it.

A popup that comes out when you try to type on a text field huh? Genius solution.

**Pull request merged**